src/BL/TransactionManager.java

```diff
@@ -0,0 +1,64 @@
+package BL;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class TransactionManager {
+    // Определяет приватное поле dataRate, предназначенное для хранения экземпляра класса CurrencyExchangeHandler
+    private final CurrencyExchangeHandler dataRate;
+    // Определяет приватное поле dataHistory, предназначенное для хранения экземпляра класса TransactionHistoryHandler
+    private final TransactionHistoryHandler dataHistory;
+
+    // Конструктор класса
+    public TransactionManager(String rateFilePath, String historyFilePath ) {
+        this.dataRate = new CurrencyExchangeHandler(rateFilePath);
+        this.dataHistory = new TransactionHistoryHandler(historyFilePath);
+
+    }
+
+    /**
+     * Метод addTransaction - добавляет новую транзакцию в историю транзакций.
+     * Параметры метода: InitialCurrencyAmount - сумма обмена, initialCurrency начальная валюта, resultCurrency  конечная валюта
+     * Использует метод read() класса TransactionHistoryHandler.
+     **/
+    public void addTransaction(double initialCurrencyAmount, CurrencyTitle initialCurrency, CurrencyTitle resultCurrency) {
+        // Формируем ArrayList из транзакций из истории
+        ArrayList<Transaction> data = this.dataHistory.read();
+        // Создаем новую транзакцию
+        //TODO - проверить конструктор в классе Transaction
+        Transaction newTransaction = new Transaction(
+                LocalDateTime.now(),
+                initialCurrencyAmount,
+                initialCurrency,
+                resultCurrency,
+                dataRate.calculateRate(initialCurrency, resultCurrency)
```

Comment:
_calculateRate_ is a method in CurrencyManager. you can create an instance of CurrencyManager in the constructor and store it in a class variable for using here